crates/uv/src/commands/project/mod.rs

```diff
@@ -1893,12 +1960,15 @@ pub(crate) async fn resolve_environment(
     let python_requirement = PythonRequirement::from_interpreter(interpreter);
 
     // Initialize the registry client.
-    let client = RegistryClientBuilder::new(client_builder, cache.clone())
+    let mut builder = RegistryClientBuilder::new(client_builder, cache.clone())
         .index_locations(index_locations.clone())
         .index_strategy(*index_strategy)
         .markers(interpreter.markers())
-        .platform(interpreter.platform())
-        .build();
+        .platform(interpreter.platform());
+
+    builder = builder.pre_download_hook_arc(pre_download_hook);
+
```

Comment:
it seems like I should just be able to do this as part of the builder. I keep banging against the compiler doing so though.